File: expensereport-csharp/expensereport-ut/ExpenseReportTest.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.IO;
using System.Threading.Tasks;
using expensereport_csharp;
using NUnit.Framework;
using VerifyNUnit;
using VerifyTests;

namespace Tests
{
    public class ExpenseReportTest
    {
        public Expense createExpense(ExpenseType type, int amount)
        {
            var expense = new Expense();
            expense.type = type;
            expense.amount = amount;
            return expense;
        }

        [Test]
        public Task ExpenseReport()
        {
            // Redirect Console output to a StringWriter
            var sw = new StringWriter();
            Console.SetOut(sw);

            var expenses = new List<Expense>()
            {
                createExpense(ExpenseType.DINNER, 5000),
                createExpense(ExpenseType.DINNER, 5001),
                createExpense(ExpenseType.BREAKFAST, 1000),
                createExpense(ExpenseType.BREAKFAST, 1001),
                createExpense(ExpenseType.CAR_RENTAL, 200)
            };
            new ExpenseReport().PrintReport(expenses);
            
            sw.Flush();
            var allWrittenLines = sw.ToString();

            var settings = new VerifySettings();
            // TODO: improve this to only scrub the date and not the whole line
            settings.ScrubLines(line => line.Contains("Expenses"));
            return Verifier.Verify(allWrittenLines, settings);
        }
    }
}

File: expensereport-php/tests/ExpenseReportTest.php
<?php

namespace Expenses\Test;

use Expenses\Expense;
use Expenses\ExpenseReport;
use Expenses\ExpenseType;
use PHPUnit\Framework\TestCase;

class ExpenseReportTest extends TestCase {

    function createExpense($type, $amount): Expense
    { return new Expense($type, $amount); }

    public function testReport()
    {
        ob_start();

        $expenses = [
            $this->createExpense(ExpenseType::DINNER, 5000),
            $this->createExpense(ExpenseType::DINNER, 5001),
            $this->createExpense(ExpenseType::BREAKFAST, 1000),
            $this->createExpense(ExpenseType::BREAKFAST, 1001),
            $this->createExpense(ExpenseType::CAR_RENTAL, 200)
        ];
        $report = new ExpenseReport();
        $report->print_report($expenses);

        $output = ob_get_clean();
        $this->assertEquals("hello world!", $output);
    }
}


